### ISSUE TITLE
clang-tidy check bugprone-unused-local-non-trivial-variable should honor [[maybe_unused]]

### ISSUE THREAD

**Issue by pkl97** [llvm/llvm-project#81419]
In clang 18.1.0-rc2 clang-tidy check bugprone-unused-local-non-trivial-variable emits a warning even if the variable in question is marked [[maybe_unused]].

A simple example:
```c++
#include <string>

int main()
{
    const std::string s1; // warning is ok
    [[maybe_unused]] const std::string s2; // no warning expected
    return 0;
}
```
Both lines trigger an "unused local variable" warning.

Honestly I think this defeats the purpose of this attribute and makes it unnecessarily hard to write platform independent code.
Flagging with NOLINT() is always a possibility however a non-clang compiler will still require the [[maybe_unused]] attribute to suppress the warning. And the presence of the attribute indicates that the developer has thought about the variable being unused in the first place.

**Comment by pkl97** [comment]
Thanks a lot for the quick fix!

Do you see any chance of a backport to the clang 18 branch? It would be great to have this feature included in the upcoming release.

**Comment by HerrCai0907** [comment]
Sorry, I am not familiar with back port policy but I remember we can only port serious flaw.

**Comment by pkl97** [comment]
Ok, I understand.

Nevertheless clang-tidy check bugprone-unused-local-non-trivial-variable is a very valuable tool that already helped us identifying a few issues in our code. Great work!
